### ISSUE TITLE
[Question] Sofia Sans Matrix file?

### ISSUE THREAD

**Issue by UwQ-Official** [lettersoup/Sofia-Sans#46]
## Do you have any idea where I can find either an `otf` or `ttf` file of Sofia Sans Matrix (used on bus/trolleybus destination signs)?

The current matrix files use `fnt` file type, that can't be opened by anything.

**Comment by kenmcd** [comment]
FontForge can open `.fnt` files.
But they are usually bitmap fonts.
Supposedly they can also be vector, but I have never seen one.

Can you post the files here? (put them in a ZIP)

**Comment by UwQ-Official** [comment]
> FontForge can open `.fnt` files.
> But they are usually bitmap fonts.
> Supposedly they can also be vector, but I have never seen one.
> 
> Can you post the files here? (put them in a ZIP)
> 

FontForge cannot open these files for some unknown reason.
It's the files located in `This Repo/SofiaSansMatrix/Font/...`.

**Comment by UwQ-Official** [comment]
Apparently there is a bad file signature?!

It would be best if somebody can provide the `src` file

**Comment by UwQ-Official** [comment]
@kenmcd Can you help?

**Comment by kenmcd** [comment]
The Readme file says the files were created with
"VISE" (VEHICLE INFORMATION SYSTEM) EDITOR-SOFTWARE
So I went and found that on the web.
Appears to be proprietary software for the **bustec** company.
https://bustec.eu/en/competencies/software/

Appears to be a non-standard format.
Cannot open with these apps (which open `.fnt` files):
- FontForge (free)
- BitFontCreator Pro (commercial)
- Hiero - Bitmap Font Tool (free)
- HB Fonteditor (free)

These files are used in their LED displays and various other electronic displays - so I am assuming that this is not a normal bitmap font file. Because they are probably providing digital info to the displays, not an actual bitmap image.
So I don't think these "fonts" are in a format which is usable.

Not sure why they are even here in the repo. 
But the demo images do look cool!

**Comment by UwQ-Official** [comment]
> The Readme file says the files were created with "VISE" (VEHICLE INFORMATION SYSTEM) EDITOR-SOFTWARE So I went and found that on the web. Appears to be proprietary software for the **bustec** company. https://bustec.eu/en/competencies/software/
> 
> Appears to be a non-standard format. Cannot open with these apps (which open `.fnt` files):
> 
> * FontForge (free)
> * BitFontCreator Pro (commercial)
> * Hiero - Bitmap Font Tool (free)
> * HB Fonteditor (free)
> 
> These files are used in their LED displays and various other electronic displays - so I am assuming that this is not a normal bitmap font file. Because they are probably providing digital info to the displays, not an actual bitmap image. So I don't think these "fonts" are in a format which is usable.
> 
> Not sure why they are even here in the repo. But the demo images do look cool!

Yes, I was able to also find that, but my question if there are any useable files / a way to download VISE (probably no way).

**Comment by kenmcd** [comment]
> Yes, I was able to also find that, but my question if there are any useable files / a way to download VISE (probably no way).

Even if you had VISE it is doubtful it would do you any good.
These appear to be data files for some hardware.
They are not fonts which could be used anywhere else.

**Comment by UwQ-Official** [comment]
Damn :(

**Comment by UwQ-Official** [comment]
I have re-made the 19px version in a free online app, but I don't think I'm allowed to share it due to copyright

**Comment by kenmcd** [comment]
> I have re-made the 19px version in a free online app, but I don't think I'm allowed to share it due to copyright

I assume you made it from the Sofia Sans OFL licensed fonts - so no problem in sharing it.

**Comment by UwQ-Official** [comment]
> > I have re-made the 19px version in a free online app, but I don't think I'm allowed to share it due to copyright
> 
> I assume you made it from the Sofia Sans OFL licensed fonts - so no problem in sharing it.

I remade this:
![image](https://github.com/user-attachments/assets/96e11f8a-fa46-425e-941c-67f8d7baf079)
⬇️ 
![image](https://github.com/user-attachments/assets/0b7b7070-2393-412b-ba57-c0189e682064)

**Comment by UwQ-Official** [comment]
I'm closing this as there is no point in getting the original at this point ┐⁠(⁠ ⁠∵⁠ ⁠)⁠┌

**Comment by UwQ-Official** [comment]
I managed to download VISE 😌
You can just request it on their website and with a little more luck they'll respond ;)

**Comment by UwQ-Official** [comment]
This is what these two `fnt` files contain:

16px:
<img width="1244" height="321" alt="Image" src="https://github.com/user-attachments/assets/d2271da8-6d0b-4efa-8d14-e7f55cd56798" />

19px:
<img width="1218" height="420" alt="Image" src="https://github.com/user-attachments/assets/081dbbb9-e2e4-4979-b529-39e9641c19ab" />

**Comment by kenmcd** [comment]
Those are kinda cool.
How were the images created?

**Comment by UwQ-Official** [comment]
> Those are kinda cool. How were the images created?

Automatically generated I guess...
<img width="1919" height="1019" alt="Image" src="https://github.com/user-attachments/assets/6332f870-2717-4d46-a83b-bb99e99c9a1b" />

**Comment by kenmcd** [comment]
That interface is quite intriguing.
What happens if you select the `Font of pictograms` box and then do Export?
Do you get some kind of actual bitmap font?
That would make it even easier to replicate.

Since they show other languages in the demo pics, I assume some of the other fonts are for Latin characters instead of just Cyrillic.
The images remind me of the Bitcount fonts, but they are only a 9 pixel grid (IIRC).
Those images would make a great pixel font.
Doing a full character set is hard with pixel fonts as you need to add all the accents, and punctuation, etc.
I only see one Cyrillic accent, and only a little punctuation.

I would really like to see this app.
If you could attach it here in a zip that would be great.
If you want, you can delete it after I grab it.

The Bitcount pixel fonts are built using a grid, and they just put the dot elements in the grid. So those images show us their pre-made grid-fit characters, so something similar could be done with those characters..

**Comment by UwQ-Official** [comment]
1. `Font of pictograms` does nothing (at least visually)
2. Export just exports the selected fonts in one of 3 formats:
    `fnt` - Font files
    `rgbf` - RGB Font file
    `bfnt` - Bustec font files
3. Fonts 0-8 are ASCII (including some letters commonly used in Czech). 10-11 are line numbers (`0-9` & `А`). Font №15 is graphics:
<img width="1246" height="392" alt="Image" src="https://github.com/user-attachments/assets/f831c560-1486-462c-89c2-b4590dc1a715" />
4. I am **not** allowed to re-distribute this software, sorry :(

4.1 I suggest you kindly ask on [bustec's website](https://bustec.eu/en/contact/contact-form/) pretending to be public transport operator staff (that's what I did)
5. You can (with some luck and knowledge) make an LED display big enough to put every letter on it and save it as an image (which is a very convenient option they give)
<img width="920" height="309" alt="Image" src="https://github.com/user-attachments/assets/2e2638cb-3f33-4539-9a62-c3df799aa115" />
This is the image (exported, converted from `bmp` to `png`)
<img width="144" height="19" alt="Image" src="https://github.com/user-attachments/assets/d781854f-a4be-45a3-8730-7c0e945954a9" />
---
Font №10 (cyrillic `А`)
<img width="901" height="124" alt="Image" src="https://github.com/user-attachments/assets/03f8a867-eaae-4d0e-90bb-7f621b2ff026" />

Font №15:
<img width="1092" height="475" alt="Image" src="https://github.com/user-attachments/assets/d47b32c2-e369-4c38-b604-3d4f7bcd93be" />

Sorry for all the weird Markdown fonts and shi-

**Comment by UwQ-Official** [comment]
You can also pick any character from a font

<img width="600" height="357" alt="Image" src="https://github.com/user-attachments/assets/a6e23db9-92c9-4659-9a97-1571859de2e3" />

**Comment by UwQ-Official** [comment]
> What happens if you select the `Font of pictograms` box and then do Export?
> Do you get some kind of actual bitmap font?

Sorry, I didn't read it carefully. I'll try this later and give you results

**Comment by UwQ-Official** [comment]
@kenmcd 

`Font of pictograms` does nothing to exported files (besides changing a single byte at the end of the file).

(`np` - Not font of pictograms; `p` - Font of pictograms)
<img width="1538" height="703" alt="Image" src="https://github.com/user-attachments/assets/05401808-b2c6-4347-a0bc-35c11bfcd902" />
